crest/Assets/Crest/Crest/Scripts/Shapes/ShapeGerstner.cs

```diff
@@ -0,0 +1,619 @@
+                var phaseOffsetMeters = Vector3.Dot(newOrigin, direction);
+
+                // wave number
+                var k = 2f * Mathf.PI / _wavelengths[i];
+
+                _phases[i] = Mathf.Repeat(_phases[i] + phaseOffsetMeters * k, Mathf.PI * 2f);
+            }
+        }
+
+        public void UpdateWaveData()
+        {
+            // Set random seed to get repeatable results
+            Random.State randomStateBkp = Random.state;
+            Random.InitState(_randomSeed);
+
+            _spectrum.GenerateWaveData(_componentsPerOctave, ref _wavelengths, ref _angleDegs);
+
+            UpdateAmplitudes();
+
+            // Won't run every time so put last in the random sequence
+            if (_phases == null || _phases.Length != _wavelengths.Length)
+            {
+                InitPhases();
+            }
+
+            Random.state = randomStateBkp;
+
+            SliceUpWaves();
+        }
+
+        void UpdateAmplitudes()
+        {
+            if (_amplitudes == null || _amplitudes.Length != _wavelengths.Length)
+            {
+                _amplitudes = new float[_wavelengths.Length];
+            }
+
+            for (int i = 0; i < _wavelengths.Length; i++)
+            {
+                _amplitudes[i] = _weight * _spectrum.GetAmplitude(_wavelengths[i], _componentsPerOctave);
+            }
+        }
+
+        void InitPhases()
+        {
+            // Set random seed to get repeatable results
+            Random.State randomStateBkp = Random.state;
+            Random.InitState(_randomSeed);
+
+            var totalComps = _componentsPerOctave * OceanWaveSpectrum.NUM_OCTAVES;
+            _phases = new float[totalComps];
+            for (var octave = 0; octave < OceanWaveSpectrum.NUM_OCTAVES; octave++)
+            {
+                for (var i = 0; i < _componentsPerOctave; i++)
+                {
+                    var index = octave * _componentsPerOctave + i;
+                    var rnd = (i + Random.value) / _componentsPerOctave;
+                    _phases[index] = 2f * Mathf.PI * rnd;
+                }
+            }
+
+            Random.state = randomStateBkp;
+        }
+
+        private void ReportMaxDisplacement()
+        {
+            if (_spectrum._chopScales.Length != OceanWaveSpectrum.NUM_OCTAVES)
+            {
+                Debug.LogError($"OceanWaveSpectrum {_spectrum.name} is out of date, please open this asset and resave in editor.", _spectrum);
+            }
+
+            float ampSum = 0f;
+            for (int i = 0; i < _wavelengths.Length; i++)
+            {
+                ampSum += _amplitudes[i] * _spectrum._chopScales[i / _componentsPerOctave];
+            }
+            OceanRenderer.Instance.ReportMaxDisplacementFromShape(ampSum * _spectrum._chop, ampSum, ampSum);
+        }
+
+        void InitBatches()
+        {
+            var registered = RegisterLodDataInputBase.GetRegistrar(typeof(LodDataMgrAnimWaves));
+
+            //#if UNITY_EDITOR
```

Comment:
Does this need to be uncommented?

Reply:
No i think it should be removed, which i've done now. I dont really recall why there are ifdefs there. I feel it is best to just remove the old batches every time, whether in editor or standalone, so i am making it do that.